core/extension/gdextension.cpp

```diff
@@ -495,17 +503,21 @@ void GDExtension::close_library() {
 	library = nullptr;
 }
 
+bool GDExtension::is_embedded() const {
+	return embedded;
+}
+
 bool GDExtension::is_library_open() const {
-	return library != nullptr;
+	return (library != nullptr) || embedded;
```

Comment:
Having to check for `embedded` at runtime in so many places doesn't feel quite right. Compiling Godot in this "LibGodot mode" is not the common case. Can all these checks be `#ifndef`'d away for when running Godot normally?